crates/red_knot_python_semantic/src/types.rs

```diff
@@ -525,6 +523,11 @@ impl<'db> Type<'db> {
                 .elements(db)
                 .iter()
                 .any(|&elem_ty| ty.is_subtype_of(db, elem_ty)),
+            (Type::Instance(self_class), Type::Instance(target_class))
+                if self_class.is_known(db, KnownClass::NoneType) =>
+            {
+                target_class.is_known(db, KnownClass::NoneType)
+            }
```

Comment:
We should be able to remove this branch too once we understand that `NoneType` is a subclass of `object` (my MRO PR is coming soon!!)

Reply:
Note that this is about `NoneType <: NoneType`. The `X <: object` case is handled further above.

I'm actually surprised that `NoneType <: NoneType` doesn't work without this. We have a `other == self` check in `is_subclass_of` on the `ClassType`s. Is that not working correctly? Is it actually checking for equal classes or for some other identity (salsa).

> my MRO PR is coming soon!!

That will definitely help us remove some cases!

Reply:
I pushed a commit that avoids this branch and handles it inside `is_sublass_of`, by also comparing on known classes. I think that `self == other` check there is a little misleading.

Reply:
`self == other` is indeed just checking salsa IDs. But Salsa IDs are assigned by Salsa interning, which interns based on struct equality, so that should be the same thing as an `Eq` check.

It would be good to understand what's happening here. Let me know if you'd like another pair of eyes on it. Maybe it's related to the fact that there are multiple `NoneType` in typeshed? (The one in `_typeshed` and the one in `types` for more recent Python versions.)

Also, I think this branch should go into `is_equivalent_to` instead of `is_subtype_of`.